insights/v1alpha2/types_insights.go

```diff
@@ -0,0 +1,410 @@
+	GathererStateEnabled GathererState = "Enabled"
+	// GathererStateDisabled gatherer state, which means that the gatherer will not run.
+	GathererStateDisabled GathererState = "Disabled"
+)
+
+// DataGatherStatus contains information relating to the DataGather state.
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.insightsRequestID) || has(self.insightsRequestID))",message="cannot remove insightsRequestID attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.startTime) || has(self.startTime))",message="cannot remove startTime attribute from status"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.finishTime) || has(self.finishTime))",message="cannot remove finishTime attribute from status"
+// +kubebuilder:validation:Optional
+type DataGatherStatus struct {
+	// conditions is an optional field that provides details on the status of the gatherer job.
+	// It may not exceed 100 items and must not contain duplicates.
+	//
+	// The current condition types are DataUploaded, DataRecorded, DataProcessed, RemoteConfigurationNotAvailable, RemoteConfigurationInvalid
+	//
+	// The DataUploaded condition is used to represent whether or not the archive was successfully uploaded for further processing.
+	// When it has a status of True and a reason of Succeeded, the archive was successfully uploaded.
+	// When it has a status of Unknown and a reason of NoUploadYet, the upload has not occurred, or there was no data to upload.
+	// When it has a status of False and a reason Failed, the upload failed and the reason reflects the returned HTTP status code. The accompanying message will include the specific error encountered.
+	//
+	// The DataRecorded condition is used to represent whether or not the archive was successfully recorded.
+	// When it has a status of True and a reason of AsExpected, the archive was recorded successfully.
+	// When it has a status of Unknown and a reason of NoDataGatheringYet, the data gathering process has not started yet.
+	// When it has a status of False and a reason of RecordingFailed, the recording failed and a message will include the specific error encountered.
+	//
+	// The DataProcessed condition is used to represent whether or not the archive was processed by the processing service.
+	// When it has a status of True and a reason of Processed, the data was processed successfully.
+	// When it has a status of Unknown and a reason of NothingToProcessYet, there is no data to process at the moment.
+	// When it has a status of False and a reason of Failure, processing failed and a message will include the specific error encountered.
+	//
+	// The RemoteConfigurationNotAvailable condition is used to represent whether the remote configuration is available.
+	// When it has a status of Unknown and a reason of Unknown, the state of the remote configuration is unknown—typically at startup.
+	// When it has a status of True and a reason of AsExpected, the configuration is available.
```

Comment:
`RemoteConfigurationNotAvailable` with a status of `True` reads like "it is true that the remote configuration is _not_ available" as opposed to what the goal of this states sounds like based on the comment.

Maybe changing the status condition name to `RemoteConfigurationAvailable` would be better here?

Reply:
Sounds good to me. WDYT @tremes

Reply:
Yes in reality it's `RemoteConfigurationAvailable`. The negation doesn't make any sense.